api/commands/run.ts

```diff
@@ -103,24 +118,29 @@ export class Runner {
     this.runResult.timing = timer.end();
 
     if (this.graph.runConfig && this.graph.runConfig.useRunCache) {
-      await this.adapter.prepareStateMetadataTable();
-
-      await this.adapter.deleteStateMetadata(this.graph.actions);
-
-      const successfulActions = this.runResult.actions
-        .filter(action =>
-          [
-            dataform.ActionResult.ExecutionStatus.SUCCESSFUL,
-            dataform.ActionResult.ExecutionStatus.CACHE_SKIPPED
-          ].includes(action.status)
-        )
-        .map(action =>
-          this.graph.actions.find(executionAction => action.name === executionAction.name)
-        );
-
-      // Currently, we don't support caching for operations (and any dependents)
-      await this.adapter.persistStateMetadata(
-        successfulActions.filter(action => action.type !== "operation")
+      await this.dbadapter.prepareStateMetadataTable();
+      await this.dbadapter.deleteStateMetadata(this.graph.actions);
+      await this.dbadapter.persistStateMetadata(
+        this.graph.actions.filter(executionAction => {
+          if (executionAction.hermeticity !== dataform.ActionHermeticity.HERMETIC) {
+            return false;
```

Comment:
If an action goes from hermetic to non hermetic, will this cause issues due to cached state data that we never overwrite?

Reply:
I don't think this is an issue, no.

If an action went to non-hermetic, we'll always run it regardless of what's in the cached state.

If it then returns to being non-hermetic, either: (a) if something else about its definition changed, the hash will break, or (b) if the table in the warehouse was modified, or any of its inputs was modified, it will not cache-skip, or (c) it must be in exactly the same state as it was when the cache record was originally written, and none of its inputs can have changed, thus it will (correctly) cache-skip.